warehouse/accounts/views.py

```diff
@@ -118,7 +86,9 @@ def login(request, redirect_field_name=REDIRECT_FIELD_NAME,
 
         # Now that we're logged in we'll want to redirect the user to either
         # where they were trying to go originally, or to the default view.
-        return HTTPSeeOther(redirect_to, headers=dict(headers))
+        return HTTPSeeOther(
+            redirect_to,
+            headers=dict(_login_user(request, userid)))
```

Comment:
Yeah, I was on the fence about that. Generally I prefer to use local variables only if they're used in multiple places (and it doesn't make the code look bad), but agreed on the side effect, so will change.